src/app/_components/Search/Search.tsx

```diff
@@ -497,42 +535,91 @@ function SearchInput({ searchTermFromQueryParams = '' }) {
   );
 }
 
-export function Search() {
+export function Search({ fullScreen = false }: { fullScreen?: boolean }) {
+  const dispatch = useAppDispatch();
   const searchTermFromQueryParams = buildAdvancedSearchQuery(useFilterParams());
   const recentResults = useRecentResultsLocalStorage();
 
   const isSearchFieldFocused = useAppSelector(selectIsSearchFieldFocused);
   const searchTerm = useAppSelector(selectSearchTerm);
   const searchResponse = useSearchQuery(searchTerm, true);
   const isLoading = searchResponse.isLoading;
+  const isMobile = useBreakpointValue({ base: true, lg: false });
```

Comment:
I think this makes this unable to be ssr. Perhaps we can take the same approach as the mobile menu and make separate the mobile logic into a wrapper around this component. Knowing that it will only be rendered on mobile allows you not to have to use the usebreakpointvalue hook.

see this code for what I am talking about /
const MobileNavBar = ({ tokenPrices, ...props }: { tokenPrices: TokenPrice } & FlexProps) => {
  const { open, onToggle } = useDisclosure();

  return (
    <>
      <SharedMobileNavBar onIconClick={onToggle} icon={<List />} {...props} />
      {open && <MobileNavPage onClose={onToggle} tokenPrices={tokenPrices} />}
    </>
  );
};

Reply:
I believe as long a component has a decision to make based on screen size, it has to be client side no matter how we set it up.

Reply:
The issue is that usebreakpointvalue could possible return a value on the server that does not match what it will be on the client, causing a hydration mismatch, multiple renders, and potential UI flashing. When the breakpoint logic is implemented with CSS the server deoesn't have to make a decision and can generate the HTML and CSS and send it to the client where the browser can apply the CSS rules. We also agreed not to use useBreakpointValue, due to these issues. That's why we took it out in the first place.